cpp/src/Ice/SSL/SSLUtil.h

```diff
@@ -44,7 +45,7 @@ namespace Ice::SSL
     void readFile(const std::string&, std::vector<char>&);
 
     // Determine if a file or directory exists, with an optional default directory.
-    bool checkPath(const std::string&, const std::string&, bool, std::string&);
+    std::optional<std::string> checkPath(const std::string&, const std::string&, bool);
```

Comment:
For new/updated code, we prefer parameter names in function signatures.

Reply:
What about splitting checkPath in checkDirPath and checkFilePath for better readability? I doubt the bool is ever a runtime variable.